### PR TITLE
Leader testing

### PR DESCRIPTION
**Pull request by changnor**

### CODE REVIEW

File: test/leader/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Leader Tests
+## Prerequisites
+1. Leader vehicle must be powered on during all testing. 
+2. Controller must be connected to vehicle for the duration of testing in order to give input.
```
Comment (johnshamoon) [inline test/leader/README.md]: Controller.

The controller should not be giving any input.

File: test/leader/leader_test.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+class LeaderTest(unittest.TestCase):
+    def test_set_speed_lower_bound(self):
+        leader = Leader()
+
+        self.assertEqual(leader.set_speed(-1), 0)
+
+
+    def test_set_speed_upper_bound(self):
+        leader = Leader()
+
```
Comment (johnshamoon) [inline test/leader/leader_test.py]: this blank space is unnecessary and makes the tests harder to follow.

File: test/leader/leader_test.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        self.assertEqual(leader.set_speed(1) , 50)
```
Comment (johnshamoon) [inline test/leader/leader_test.py]: set speed doesn't return anything.

Comment (johnshamoon) [reply test/leader/leader_test.py]: don't put a space between the parenthesis and the comma

Comment (johnshamoon) [reply test/leader/leader_test.py]: And why 50? What if I change the max speed?

File: test/leader/leader_test.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+    def test_set_speed_zero(self):
+        leader = Leader()
+
+        self.assertEqual(leader.set_speed(0) , 50/2)
```
Comment (johnshamoon) [inline test/leader/leader_test.py]: again, why 50?

File: test/leader/leader_test.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    def test_reverse_zero_speed(self):
+        leader = Leader()
+
+        self.assertEqual(leader.set_speed(-1) , 0)
+
+
+if __name__ == '__main__':
+    unittest.main()
+
```
Comment (johnshamoon) [inline test/leader/leader_test.py]: fix whatever happened here too

File: test/leader/leader_test.py (hunk 2 above)
Comment (changnor) [reply test/leader/leader_test.py]: Should I remove all the blank spaces before the asserts? I followed Wisam's style, should I not?

File: test/leader/leader_test.py (hunk 3 above)
Comment (changnor) [reply test/leader/leader_test.py]: I followed the test case specification, the input and desired output was specified.

Comment (changnor) [reply test/leader/leader_test.py]: instead of 50 i'll just put leader.MAX_SPEED

File: test/leader/README.md (hunk 1 above)
Comment (changnor) [reply test/leader/README.md]: modified to driver gives input

File: test/leader/leader_test.py (hunk 2 above)
Comment (changnor) [reply test/leader/leader_test.py]: I removed the blank spaces, does that look better?

File: test/leader/leader_test.py (hunk 5 above)
Comment (changnor) [reply test/leader/leader_test.py]: Its not complaining anymore

**Pull request closed without merging**